### ISSUE TITLE
v-on-handler-style: False positive for ["method", "inline"]

### ISSUE THREAD

**Issue by ilyub** [vuejs/eslint-plugin-vue#2049]
**Checklist**

- [+] I have tried restarting my IDE and the issue persists.
- [+] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.28.0
- **eslint-plugin-vue version:** 9.7.0
- **Node version:** 18.12.1
- **Operating System:** Windows 10

**Please show your full configuration:**

```js
"vue/v-on-handler-style": ["warn", ["method", "inline"]]
```

**What did you do?**	

```vue
<my-comp @update:model-value="$emit('update:modelValue', $event)">
```

**What did you expect to happen?**

No error, as in your samples:
```vue
  <!-- ✓ GOOD -->
  <button v-on:click="handler" />
  <button v-on:click="handler() /* comment */" />
```

**What actually happened?**

Getting "Prefer method handler over inline handler in v-on."

**Repository to reproduce this issue**

I'll create repo on demand or will use [playground](https://ota-meshi.github.io/eslint-plugin-vue-demo/) if you add an option to configure rules.

**Comment by FloEdelmann** [comment]
I'd argue the lint error is correct, as `$emit(…)` is actually inline style, and it could be replaced by method style instead:

```vue
<template>
  <my-comp @update:model-value="updateModelValue">
</template>

<script setup>
const emit = defineEmits(['update:modelValue']);
const updateModelValue = payload => {
  emit('update:modelValue', payload);
});
</script>

**Comment by ilyub** [comment]
But in your own example:
```ts
<!-- ✓ GOOD -->
<button v-on:click="handler" />
<button v-on:click="handler() /* comment */" />
```
"handler()" can be transformed to "handler" and yet it is marked as good.

**Comment by FloEdelmann** [comment]
Only because there is a comment.

**Comment by FloEdelmann** [comment]
I think you are referring to [this example](https://eslint.vuejs.org/rules/v-on-handler-style.html#method-inline-ignoreincludescomment-true)?

![grafik](https://user-images.githubusercontent.com/202916/203968931-993f7dd7-55cc-4581-8669-bbe1250ac4ae.png)

There, `handler()` is only allowed if followed by a comment, and that's only because the `ignoreIncludesComment` rule option is set to `true` (`false` by default).

**Comment by jacekkarczmarczyk** [comment]
> "handler()" can be transformed to "handler"

No, it can't (in some cases)

```
function handler (value = undefined) {
  ...
}
```

`@event="handler"` - handler will be called with `$event`
`@event="handler()` - handler will be called with `undefined`

**Comment by FloEdelmann** [comment]
@jacekkarczmarczyk But you could add another method:

```vue
<template>
  <my-comp @event="callHandler">
</template>

<script setup>
function handler(value = undefined) {
  // ...
}
function callHandler() {
  handler()
}
</script>
```

If you don't like that, you could opt to either disable the rule, or set it do `"inline"` or `"inline-function"` directly, since those styles are much more versatile than the `"method"` style.

**Comment by ilyub** [comment]
I think what mislead me is that I thought that ["method", "inline"] means either method or inline.
I see your logic that "$emit('update:modelValue', $event)" can be moved from template to script part.

**Comment by ota-meshi** [comment]
Feel free to open a PR if you think the documentation is missing.

**Comment by mesqueeb** [comment]
I'm sorry to comment on a closed issue.
I'm a bit confused on the Default config myself: `["method", "inline-function"]`

I would _expect_ both `method` and `inline-function` to be allowed. But it seems to want `method` and only allow `inline-function` in a v-for.

How can we allow for either `method` or `inline-function` at all times?

Or should I open a new issue to request a `"method-or-inline-function"` option?

**Comment by FloEdelmann** [comment]
Yes, please open a new issue.
